### ISSUE TITLE
torch.norm does not broadcast for dim > 2

### ISSUE THREAD

**Issue by lhk** [pytorch/pytorch#44552]
## 📚 Documentation
[From the docs:](https://pytorch.org/docs/stable/generated/torch.norm.html)

`torch.norm(input, p='fro', dim=None, keepdim=False, out=None, dtype=None)`

> **dim** (int, 2-tuple of python:ints, 2-list of python:ints, optional) – [...] If the input tensor has more than two dimensions, the vector norm will be applied to last dimension.

The docs say that by default the norm will be applied to the last dimension, when the input tensor has more than two axes. I see different behaviour:
```
a = torch.rand(3, 4, 5)
torch.norm(a)
# tensor(4.7850)
```

**Comment by Ipsit1234** [comment]
@lhk  I am not able to reproduce the error. Can you give more details?

**Comment by mariosasko** [comment]
I think the docs should explain a bit better what happens when the input is a multi-dimensional tensor. IMHO the line `If the input tensor has more than two dimensions, the vector norm will be applied to last dimension` is somewhat misleading. 

I assume they follow the explanation from numpy, where they don't bother with the multi-dim case. I like this explanation from tensorflow `If axis is None (the default), the input is considered a vector and a single vector norm is computed over the entire set of values in the tensor` so it makes sense to add a similar explanation to PyTorch.

@mruberry If you agree, I can submit a PR with the change.

**Comment by mruberry** [comment]
Fixing the existing docs sounds good. Do you think this is correctly documented for the newer [torch.linalg.norm](https://pytorch.org/docs/master/linalg.html?highlight=norm#torch.linalg.norm)?

**Comment by mariosasko** [comment]
Yes, there is this line in the `torch.linalg.norm` docs: `If both dim and ord are None, the 2-norm of the input flattened to 1-D will be returned` so it's clear how the function behaves in the multi-dim case.

**Comment by mruberry** [comment]
Great. I thought the torch.linalg.norm doc was much clearer, too. What about porting language like that to torch.norm?

**Comment by mariosasko** [comment]
Yes, will try to do just that.
